### PR TITLE
MGMT-3519 Change control plane replicas patch to do the patch only when control plane replicas is 3

### PR DESCRIPTION
**Pull request by ori-amizur**
- This fixes it for IPv6 single node

### CODE REVIEW

File: src/installer/installer.go [hunk 1]
```diff
@@ -307,19 +307,28 @@ func (i *installer) waitForControlPlane(ctx context.Context, kc k8s_client.K8SCl
...
+		if origControlPlaneReplicas, err = kc.GetControlPlaneReplicas(); err != nil {
+			i.log.WithError(err).Error("Failed to get control plane replicas")
 			return err
 		}
+		if origControlPlaneReplicas == 3 {
+			if err = kc.PatchControlPlaneReplicas(); err != nil {
+				i.log.WithError(err).Error("Failed to patch control plane replicas")
+				return err
+			}
+		}
```
Comment (eranco74) [inline src/installer/installer.go:321]: Extract to a separate function

File: src/installer/installer.go [hunk 2]
```diff
@@ -307,19 +307,28 @@ func (i *installer) waitForControlPlane(ctx context.Context, kc k8s_client.K8SCl
...
 	}
+	var origControlPlaneReplicas int
 	if nt == ovnKubernetes {
-		if err = kc.PatchControlPlaneReplicas(); err != nil {
-			i.log.WithError(err).Error("Failed to patch control plane replicas")
+		if origControlPlaneReplicas, err = kc.GetControlPlaneReplicas(); err != nil {
+			i.log.WithError(err).Error("Failed to get control plane replicas")
 			return err
 		}
+		if origControlPlaneReplicas == 3 {
```
Comment (eranco74) [inline src/installer/installer.go:316]: Use a const for 3.
Also, add a comment on why it's required.

Comment (ori-amizur) [reply src/installer/installer.go:316]: OK

File: src/installer/installer.go (hunk 1 above)
Comment (ori-amizur) [reply src/installer/installer.go:321]: OK

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**